tmt/steps/prepare/prepare.py

```diff
@@ -0,0 +1,158 @@
+import dataclasses
+import re
+from typing import List, Optional
+
+import tmt
+import tmt.base
+import tmt.log
+import tmt.options
+import tmt.steps
+import tmt.steps.prepare
+import tmt.utils
+from tmt.steps.provision import Guest
+from tmt.utils import Command, field
+
+
+class FeatureBase(tmt.utils.Common):
```

Comment:
I'm not sure what is the purpose of `FeatureBase` class. At first, I expected it to serve as a base for more specific classes, like `Epel` or `CRB`, but `FeatureBase` already has `enable_epel` & `disable_epel` methods, so that does not seem to be the case.

I have an old draft in my git stash, when I thought about working on this, I sort of had a few levels of classes, something that might turn into plugins for features (like we have for export formats or for post-test checks), something like the following:

```python
class Feature:
  KEY: str

class ToggleableFeature(Feature):
  def enabled(self, guest, logger):
    ...
  def disable(self, guest, logger):
    ...

class EPEL(ToggleableFeature):
  KEY = 'epel'

  def enabled(self, guest, logger):
    # magic...

class CRB(ToggleableFeature):
  KEY = 'crb'

  ...

_FEATURES = {
  EPEL.KEY: EPEL,
  CRB.KEY: CRB,
  ...
}

# And some day, we might turn ^^ into plugins like we have for export or Soon (TM) for post-test checks.

# in plugin's go:
for key, value in self.data.items():
    if key in some_well_known_keys_like_name_and_how:
      continue

    if key not in _FEATURES:
      raise "Unknown key"

    feature = _FEATURES[key]()

    if isinstance(feature, ToggleableFeature):
        value = value.lower()

        if value == 'enabled':
            feature.enabled(guest, logger)

        elif value == 'disabled':
            feature.disabled(guest, logger)

        else:
            raise ...
```